Release Signer Alongside Node

BowTiedDevOps opened the pull request:
The aim of this PR is to create a signer release every time there is a core release happening, with the tag set as `<core_release>.0`.

**Example workflows (test failures are unaffected by these changes):**

- Node Release (built both `5.0.0.0.0` and `signer-5.0.0.0.0.0` releases): https://github.com/BowTiedDevOps/stacks-core/actions/runs/12953974660
- Standalone Signer Release (built `signer-5.0.0.0.0.1` release): https://github.com/BowTiedDevOps/stacks-core/actions/runs/12957741282
- `develop` branch CI run: https://github.com/BowTiedDevOps/stacks-core/actions/runs/12960606782

**Useful Links:**

- Releases Page: https://github.com/BowTiedDevOps/stacks-core/releases
- Stacks Core Docker Images: https://hub.docker.com/r/bowtieddevops/stacks-core/tags
- Stacks Signer Docker Images: https://hub.docker.com/r/bowtieddevops/stacks-signer/tags

CharlieC3 reviewed (changes requested):
Mostly LGTM! However can you update the [README](https://github.com/stacks-network/actions/blob/feat/release-signer-alongside-node/stacks-core/release/README.md) first?

Some general feedback, the stacks-core/release dir is pretty difficult to follow. The semantics between actions like "create-docker-image" and "docker-images", and other actions with a similar problem, is not distinct enough. Are some of these actions intended to be internal while others are more public-facing? If so, it may help to explain that in the dir's README, or to rename/move them with that in mind.

BowTiedDevOps commented:
@CharlieC3 Thank you for the feedback! I've updated the README to further explain what the actions do - they are tailored to the `stacks-core` repository and should not be re-used for other repositories, I've also changed the placement of the referenced actions to reflect the order in which they're being called - hopefully this way, the flow makes more sense.

CharlieC3 reviewed (approved):
Thanks for updating the README and ordering it! It's much clearer now how they should be used.

fabergat commented:
LGTM, I liked the README thing.
